bridges/php-local/LocalBridge/FileManagerApi.php

```diff
@@ -343,16 +345,30 @@ private function changePermissionsAction($paths, $permissions, $recursive)
 
 	private function compressAction($paths, $destination, $archiveName)
 	{
-		$archivePath = $this->basePath . $destination . $archiveName;
-
-		$zip = new ZipArchive();
-		if ($zip->open($archivePath, ZipArchive::CREATE) !== true) return false;
-
-		foreach ($paths as $path) {
-			$zip->addFile($this->basePath . $path, basename($path));
-		}
-
-		return $zip->close();
+		$pathToZip = $this->basePath . $destination . '/' . $archiveName . '.zip';
+        $zip = new \ZipArchive();
+        if ($zip->open($pathToZip, \ZipArchive::CREATE) !== true) return false;
+        
+        foreach ($paths as $path){
+            $pathToElement = $this->basePath . $path;
+            if (is_dir($pathToElement) === true){
+                $files = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($pathToElement), \RecursiveIteratorIterator::LEAVES_ONLY);
+                foreach ($files as $file){
+                    if( in_array(substr($file, strrpos($file, '/')+1), array('.', '..')) )
+                            continue;
+                    $beautifulPath = str_replace($this->basePath . $destination,'',$file);
+                    if ($beautifulPath[0] == '/'){
+                        $beautifulPath = substr($beautifulPath,1);
+                    }
+                    $zip->addFile($file,$beautifulPath);
+                }
+            }
+            else if (is_file($pathToElement) === true)
+            {
+                $zip->addFromString(basename($pathToElement), file_get_contents($pathToElement));
```

Comment:
Great commit but why do you use addFromString and file_get_contents? You have to load to memory whole files.